curriculum/challenges/english/25-front-end-development/lab-movie-review-page/67ef8f695db52583424bd118.md

```diff
@@ -0,0 +1,176 @@
+# demoType can either be 'onClick' or 'onLoad'. If the project or lab doesn't have a preview, delete the property
+demoType: onClick
+---
+
+# --description--
+
+You'll practice concepts like semantic HTML, `alt` attributes, accessible lists, and hiding decorative content from screen readers using `aria-hidden`.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. Your should have a `main` element.
+2. Inside the `main` element, you should have an `h1` element for the movie title.
+3. Below the `h1` element, you should have an `img` element displaying the movie cover with a descriptive `alt` attribute describing the image.
+4. You should have a `p` element containing a brief movie description.
+5. You should have another `p` element to display the movie rating.
+6. Inside the `p` element, you should have a `strong` element with the text `Movie Rating`.
+7. You should have a `span` element with `aria-hidden` attribute set to `true` containing a visual representation of the rating using stars and a numerical value in parentheses (e.g. `9.2/10`).
+8. You should have a `h2` element with the text `Cast Members`.
+9. You should have a `ul` element.
+10. Inside the `ul` element, you should have multiple `li` elements each containing an actor’s name wrapped in `strong` element and their corresponding character role (e.g., `James Holloway as Ethan Carter`).
+
+# --hints--
+
+You should have a `main` element.
+
+```js
+assert.isNotNull(document.querySelector("main"));
+```
+
+Inside the `main` element, you should have an `h1` element for the movie title.
+
+```js
+const h1 = document.querySelector("main > h1");
+assert.isNotNull(h1);
+```
+
+Below the `h1` element, you should have an `img` element with a descriptive `alt` attribute.
+  
+```js
+const img = document.querySelector("main > h1 + img");
+assert.isNotNull(img);
+assert.isNotEmpty(img.getAttribute("alt"));
+```
+
+You should have a `p` element after the image that contains the movie description.
+
+```js
+const description = document.querySelector("main img + p");
+assert.isNotNull(description);
+assert.isNotEmpty(description.innerText);
+```
+
+You should have another `p` element that contains the rating.
+
+```js
+const paragraphs = document.querySelectorAll("main p");
+assert.isAtLeast(paragraphs.length, 2);
+```
+
+Inside the second `p` element, you should have a `strong` element with the text `Movie Rating`.
+
+```js
+const strongEl = document.querySelector("main p strong");
+assert.match(strongEl?.innerText.trim(), /^Movie Rating:?$/);
+```
+
+There should be a `span` element inside the rating paragraph with `aria-hidden` set to `true`.
+
+```js
+const spanEl = document.querySelector("main p span");
+assert.strictEqual(spanEl?.getAttribute("aria-hidden"), "true");
+assert.match(spanEl?.innerText, /★|☆|\d+\.\d+\/10/);
+```
+
+You should have an `h2` element with the text `Cast Members`.
```

Comment:
I would require the `h2` to come after the `h1`. Currently you can place it wherever you want.

Reply:
There is already a test for `h1` before, and the user story states that `h2` comes after the `p` element.

Reply:
you are not testing the position of the `h2` anyway